opentelemetry-api/src/opentelemetry/context/unified_context.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from opentelemetry.distributedcontext import DistributedContext
+from opentelemetry.trace import SpanContext
```

Comment:
As I understand it, this is the reason we handle the context in the trace package now. If we think of distributed tracing as being built on top of context propagation then the context package should be ignorant of tracing. This seems to be the motivation for having the tracer inject/extract the `SpanContext` from the carrier in OpenTelemetry instead of passing it into the tracer.

Reply:
Which reminds me, there was mention in the past of possibly using such context layer independently of tracing/metrics :)